Kentico.Kontent.Delivery/DeliveryClient.cs

```diff
@@ -27,6 +28,7 @@ internal sealed class DeliveryClient : IDeliveryClient
         internal readonly IPropertyMapper PropertyMapper;
         internal readonly IRetryPolicyProvider RetryPolicyProvider;
         internal readonly IDeliveryHttpClient DeliveryHttpClient;
+        internal readonly IDeliveryCacheManager DeliveryCacheManager;
```

Comment:
I would strongly recommend sticking to the Decorator pattern that was introduced in https://github.com/Kentico/kontent-boilerplate-net/blob/master/src/content/Kentico.Kontent.Boilerplate/Caching/Webhooks/CachingDeliveryClient.cs

It'll help us keep the responsibilities of classes separate.

Here's a suggested way of doing it with .NET Core's DI: https://greatrexpectations.com/2018/10/25/decorators-in-net-core-with-dependency-injection

Reply:
In the current version of the SDK is not possible to use this patter.

We installed all default dependency in the ServiceCollection extensions - you can rewrite them in separate registration. If we should use the Decorator pattern we don't very clear way how to do it with the actual architecture of the SDK.

Because things like caching and policy we can set up in the Configuration - Options and we cannot set up a dependency, which depends on this configuration.

Now we can control behavior by the Configuration - Options just in implementations classes like IDeliveryClient,..